### PR TITLE
Add soda-player latest

### PR DESCRIPTION
**Pull request by 13k**
*If there’s a checkbox you can’t complete for any reason, that's okay, just explain in detail why you weren’t able to do so.*

After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked the cask was not already refused in [closed issues].
- [x] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/caskroom/homebrew-cask/pulls
[closed issues]: https://github.com/caskroom/homebrew-cask/issues?q=is%3Aissue+is%3Aclosed
[the correct repo]: https://github.com/caskroom/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask

### CODE REVIEW

**Review by miccal — APPROVED** [review]

**Comment by 13k** [comment]
The app's [changelog page](http://changelog.sodaplayer.com) seems to be useful to track changes, I feel like it's useful information to have on the cask.

Is there any recommendation as to where mention it or any stanza in the cask that supports this?

**Review by vitorgalvao — APPROVED** [review]

File: Casks/soda-player.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+cask 'soda-player' do
+  version '1.1.2'
+  sha256 'd7ba85c4afa1631f1df809b34da7e79193b88ecfbd2961d9c2f0f92ec8f9db8a'
+
+  url 'https://releases.sodaplayer.com/mac/Soda%20Player.dmg'
+  appcast 'http://changelog.sodaplayer.com/',
+          checkpoint: '9dee7ffbaa9330c8909e950a4f51e167e5327ceea4cbbaa54d7b5cdc5d6999e8'
```
Comment (miccal) [inline Casks/soda-player.rb]: @vitorgalvao I tried that as an appcast, but `curl -s http://changelog.sodaplayer.com/` reports 404.

Comment (13k) [inline Casks/soda-player.rb]: http server is serving those based on the "Accept" header, this works: `curl -H "Accept: text/html" "http://changelog.sodaplayer.com/"`

Comment (miccal) [inline Casks/soda-player.rb]: @13k yes, but the way the `appcast` is handled via Cask reports 404.

Comment (vitorgalvao) [inline Casks/soda-player.rb]: @miccal That’s weird. `brew cask _appcast_checkpoint --calculate http://changelog.sodaplayer.com/` works flawlessly on my machine.

File: Casks/soda-player.rb [hunk 2]
```diff
@@ -1,8 +1,10 @@
 cask 'soda-player' do
-  version :latest
-  sha256 :no_check
+  version '1.1.2'
+  sha256 'd7ba85c4afa1631f1df809b34da7e79193b88ecfbd2961d9c2f0f92ec8f9db8a'
 
   url 'https://releases.sodaplayer.com/mac/Soda%20Player.dmg'
```
Comment (13k) [inline Casks/soda-player.rb]: the actual URL for versioned releases is `https://releases.sodaplayer.com/mac/Soda%20Player.dmg?version=<version>`

`1.1.3` was just released and the current cask is downloading it, then matching against the 1.1.2 hash

**Comment by 13k** [comment]
`audit --download` is failing, probably because of what @miccal and I said regarding the appcast URL returning 404.

```
brew cask audit --download soda-player
==> Downloading https://releases.sodaplayer.com/mac/Soda%20Player.dmg?version=1.1.3
######################################################################## 100.0%
==> Verifying checksum for Cask soda-player
audit for soda-player: warning
 - unexpected HTTP response code retrieving appcast: 404
Error: audit failed for casks: soda-player
```

`brew cask _appcast_checkpoint --calculate` seems to be using `Hbc::DSL::Appcast#calculate_checkpoint` and it "works", but only because `curl` returns 0 even on 404's. It's then generating a digest of the 404 HTML body returned by the server. To make curl return an error, you need to use the `-f` (`--fail`) flag.

Meanwhile, `audit` uses `Hbc::Audit#check_appcast` and it explicitly checks for HTTP status codes, so it fails appropriately.

Regardless of the above, I'm not sure how to solve the initial issue, which is to pass an `Accept` header to the appcast URL request.

File: Casks/soda-player.rb (hunk 1 above)
Comment (miccal) [inline Casks/soda-player.rb]: @vitorgalvao yes I noticed that too, but it fails `audit`.

**Comment by OranShuster** [comment]
why not just use [http://changelog.sodaplayer.com/](url)

**Comment by miccal** [comment]
I have removed the `a--cast` for the moment.

**Pull request merged**